src/Storage/Wrappers/KeyVaultClientWrapper.cs

```diff
@@ -17,31 +14,22 @@ namespace Altinn.Platform.Storage.Wrappers
     [ExcludeFromCodeCoverage]
     public class KeyVaultClientWrapper : IKeyVaultClientWrapper
     {
-        private readonly string _connectionString;
-
         /// <summary>
-        /// Initializes a new instance of the <see cref="KeyVaultClientWrapper"/> class with a client using the credentials from the key vault settings.
+        /// Initializes a new instance of the <see cref="KeyVaultClientWrapper"/> class.
         /// </summary>
-        /// <param name="keyVaultSettings">
-        /// The <see cref="KeyVaultSettings"/> with information about the principal to use when getting secrets from a key vault.
-        /// </param>
-        public KeyVaultClientWrapper(IOptions<KeyVaultSettings> keyVaultSettings)
+        public KeyVaultClientWrapper()
         {
-            _connectionString = $"RunAs=App;AppId={keyVaultSettings.Value.ClientId};" +
-                                $"TenantId={keyVaultSettings.Value.TenantId};" +
-                                $"AppKey={keyVaultSettings.Value.ClientSecret}";
         }
 
         /// <inheritdoc/>
         public async Task<string> GetSecretAsync(string vaultUri, string secretId)
         {
-            // The AzureServiceTokenProvider has its own internal static collection of principals based on the connection string.
-            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider(_connectionString);
-            KeyVaultClient keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
+            // Credentials are set based on environment variables set in Program.cs
+            SecretClient secretClient = new(new Uri(vaultUri), new DefaultAzureCredential());
```

Comment:
Could we cache the client per app owner instead of creating a new client for each request?

Reply:
We're already caching the value being retrieved.

Reply:
ah in the SasTokenProvider, got it!